runtime/vm/dllsup.c

```diff
@@ -155,3 +155,10 @@ UDATA loadJ9DLL(J9JavaVM * vm, J9VMDllLoadInfo* info) {
 	}
 }
 
+void setErrorJ9dll(J9PortLibrary *portLib, J9VMDllLoadInfo *info, const char *error, BOOLEAN errorIsAllocated) {
+	info->fatalErrorStr = error;
+	if (NULL != error && errorIsAllocated) {
+		info->loadFlags |= FREE_ERROR_STRING; /* indicates that buffer should be freed later */
+	}
+}
+
```

Comment:
Our convention is to have the return type and the opening brace on lines by themselves:
```c
void
setErrorJ9dll(J9PortLibrary *portLib, J9VMDllLoadInfo *info, const char *error, BOOLEAN errorIsAllocated)
{
```
This is missing the part that frees an existing, allocated string.

Reply:
Am I understanding correctly that I need to add a case to check whether `info->fatalErrorStr` already points to a heap-allocated string that would need to be freed before we can set it again?

Something like:
```cpp
void
setErrorJ9dll(J9PortLibrary *portLib, J9VMDllLoadInfo *info, const char *error, BOOLEAN errorIsAllocated)
{
        /* If fatalErrorStr already points to a heap-allocated string, we need to free it before we can point it to a new string */
        if (NULL != info->fatalErrorStr && (info->loadFlags & FREE_ERROR_STRING)) {
	        j9mem_free_memory(info->fatalErrorStr);
		info->loadFlags &= ~FREE_ERROR_STRING;
        }

	info->fatalErrorStr = error;
	if (NULL != error && errorIsAllocated) {
		info->loadFlags |= FREE_ERROR_STRING; /* indicates that buffer should be freed later */
	}
}
```

Reply:
Yes, like that, but paying attention to our coding standards, by adding missing parentheses and using `J9_ARE_ANY_BITS_SET()` to check if the flag is set. I also expect a cast is needed in the free call.